elements/lisk-p2p/src/p2p_types.ts

```diff
@@ -165,14 +195,17 @@ export interface P2PCompatibilityCheckReturnType {
 }
 
 export type P2PCheckPeerCompatibility = (
-	headers: P2PDiscoveredPeerInfo,
+	headers: P2PPeerInfo,
 	nodeInfo: P2PNodeInfo,
 ) => P2PCompatibilityCheckReturnType;
 
 // This is a representation of the inbound peer object according to the current protocol.
 // TODO later: Switch to LIP protocol format.
+// TODO: Include peerId as field
 export interface ProtocolPeerInfo {
```

Comment:
Please, reuse the new types or extend them to have a better understanding of what we can remove in the near future. Review the comments and place all the candidates to remove at the bottom for example.

Reply:
I have tried to reuse `P2PSharedPeerInfo` as much as I could to shorten `P2PNodeInfo` and `ProtocolNodeInfo`, couldn't update `ProtocolPeerInfo` because most of its fields are optional and TS throws an error as `P2PSharedPeerInfo` has some mandatory values